evaluate/report/csv.go

```diff
@@ -124,3 +82,228 @@ func (e *EvaluationFile) WriteEvaluationRecord(model model.Model, language langu
+		rows = append(rows, row)
+	}
+
+	return rows
+}
+
+// EvaluationRecordsPerModel holds the collection of evaluation records per model.
+type EvaluationRecordsPerLanguagePerModel map[string]EvaluationRecordsPerModel
+
+// GroupByLanguageAndModel groups the evaluation records by language and model.
+func (e EvaluationRecords) GroupByLanguageAndModel() EvaluationRecordsPerLanguagePerModel {
+	perLanguageAndModel := map[string]EvaluationRecordsPerModel{}
+
+	for _, record := range e {
+		perModel, ok := perLanguageAndModel[record.LanguageID]
+		if !ok {
+			perLanguageAndModel[record.LanguageID] = EvaluationRecordsPerModel{
+				record.ModelID: record,
+			}
+		} else {
+			_, ok := perModel[record.ModelID]
+			if !ok {
+				perModel[record.ModelID] = record.Clone()
+			} else {
+				perModel[record.ModelID].Assessments = metrics.Merge(perModel[record.ModelID].Assessments, record.Assessments)
+			}
+		}
+	}
+
+	return perLanguageAndModel
+}
+
+// loadEvaluationRecords reads and returns the evaluation records from the evaluation CSV file.
+func loadEvaluationRecords(evaluationFilePath string) (evaluationRecords EvaluationRecords, err error) {
+	evaluationFile, err := os.Open(evaluationFilePath)
+	if err != nil {
+		return nil, pkgerrors.WithStack(err)
+	}
+	defer evaluationFile.Close()
+
+	reader := csv.NewReader(evaluationFile)
+
+	// Check if the evaluation CSV header is correct.
+	if header, err := reader.Read(); err != nil {
+		return nil, pkgerrors.Wrap(err, "found error while reading evaluation file")
+	} else if strings.Join(header, ",") != strings.Join(evaluationHeader(), ",") {
+		return nil, pkgerrors.WithStack(pkgerrors.Errorf("expected header %+v\nfound header %+v", evaluationHeader(), header))
+	}
+
+	// Read the raw records from the evaluation CSV file.
+	records, err := reader.ReadAll()
+	if err != nil {
+		return nil, pkgerrors.WithStack(err)
+	}
+
+	// Convert the raw records into assessments that can be easily manipulated.
+	evaluationRecords = EvaluationRecords{}
+	for _, record := range records {
+		evaluationRecord, err := convertRawRecordToEvaluationRecord(record)
+		if err != nil {
+			return nil, err
+		}
+		evaluationRecords = append(evaluationRecords, evaluationRecord)
+	}
+
+	return evaluationRecords, nil
+}
+
+// convertRawRecordToEvaluationRecord converts a raw CSV record into an evaluation record.
+func convertRawRecordToEvaluationRecord(raw []string) (record *EvaluationRecord, err error) {
+	assessments := metrics.NewAssessments()
+
+	modelID := raw[0]
+	modelName := raw[1]
+	modelCost, err := strconv.ParseFloat(raw[2], 64)
+	if err != nil {
+		return nil, pkgerrors.WithStack(err)
+	}
+
+	languageID := raw[3]
+
+	rawMetrics := raw[7:]
+	for i, assessementKey := range metrics.AllAssessmentKeysStrings {
+		metric, err := strconv.ParseUint(rawMetrics[i], 10, 64)
+		if err != nil {
+			return nil, pkgerrors.WithStack(err)
+		}
+
+		assessments[metrics.AssessmentKey(assessementKey)] = metric
+	}
```

Comment:
nitpick, would be nice to check if the length matches :see_no_evil:

Reply:
I don't think we need it, the `reader.ReadAll()` in  `loadEvaluationRecords` already checks it the records have the length.